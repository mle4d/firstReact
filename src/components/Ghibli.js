import React from 'react';
import PropTypes from 'prop-types';

function Ghibli({ Character, Color, Movie, hex }) {
  return (
    <dl>
      <dt>Character</dt>
      <dd>{Character}</dd>
      <dt>Color</dt>
      <dd style={{ color: hex }}>{Color}</dd>
      <dt>Movie</dt>
      <dd>{Movie}</dd>
    </dl>
  );
}

Ghibli.propTypes = {
  Character: PropTypes.string.isRequired,
  Color: PropTypes.number.isRequired,
  Movie: PropTypes.string.isRequired,
  hex: PropTypes.string.isRequired
};

export default Ghibli;
