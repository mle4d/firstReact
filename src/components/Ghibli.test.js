import React from 'react';
import { shallow } from 'enzyme';
import Ghibli from './Ghibli';

describe('App component', () => {
  it('renders App', () => {
    const wrapper = shallow(<Ghibli />);
    expect(wrapper).toMatchSnapshot();
  });
});
