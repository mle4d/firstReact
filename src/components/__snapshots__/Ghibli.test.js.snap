// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<dl>
  <dt>
    Character
  </dt>
  <dd />
  <dt>
    Color
  </dt>
  <dd
    style={
      Object {
        "color": undefined,
      }
    }
  />
  <dt>
    Movie
  </dt>
  <dd />
</dl>
`;
